virttest/utils_test/libvirt.py

```diff
@@ -1059,23 +1059,39 @@ def pre_pool(self, pool_name, pool_type, pool_target, emulated_image,
                                                                nfs_path)
         elif pool_type == "iscsi":
             ip_protocal = kwargs.get('ip_protocal', "ipv4")
+            iscsi_chap_user = kwargs.get('iscsi_chap_user', None)
+            iscsi_chap_password = kwargs.get('iscsi_chap_password', None)
+            iscsi_secret_usage = kwargs.get('iscsi_secret_usage', None)
             if ip_protocal == "ipv6":
                 ip_addr = "::1"
             else:
                 ip_addr = "127.0.0.1"
-            setup_or_cleanup_iscsi(is_setup=True,
-                                   emulated_image=emulated_image,
-                                   image_size=image_size,
-                                   portal_ip=ip_addr)
+            if iscsi_chap_user and iscsi_chap_password and iscsi_secret_usage:
+                logging.debug("setup iscsi pool with chap authentication")
+                setup_or_cleanup_iscsi(is_setup=True,
```

Comment:
I see the check for chap username and password in iscsi module so we can avoid the check here but if user misconfigure credentials then appropriate authentication error will be raised, also for debug info we can have debug log in `iscsi_secret_usage` condition so by default it creates iscsi pool without authentication.